pyuvsim/telescope.py

```diff
@@ -37,3 +40,243 @@ def __eq__(self, other):
                 and self.name == other.name
             )
         return False
+
+
+class BeamList(object):
+    """
+    A container for the set of beam models and related parameters.
+
+    Each rank in the simulation gets a copy of the set of beam objects
+    used for calculating Jones matrices. Rather than broadcasting the objects
+    themselves, the list is composed of strings which provide either a complete
+    description of an analytic beam or the path to a beamfits file. After the
+    broadcast, the beams are initialized from these strings.
+
+    This class provides methods to transform between the string and object
+    representations, and also carries parameters used globally by all UVBeams,
+    including their frequency / angular interpolation options.
+
+    This behaves just as a normal Python list in terms of indexing, appending, etc.
+
+    Attributes
+    ----------
+    string_mode : bool
+        Is True if the beams are represented as strings, False if they're objects
+
+    uvb_params : dict
+        Set of additional attributes to set on UVBeam objects.
+
+    Parameters
+    ----------
+    beam_list : list (optional)
+        A list of UVBeam/AnalyticBeam objects or strings describing beams. If
+        beam_list consists of objects, then the BeamList will be initialized in
+        object mode with string_mode == False. If beam_list consists of strings,
+        the BeamList will be initialized in string mode.
+
+        Passing in a mixture of strings and objects will error.
+
+    uvb_params : dict (optional)
+        Options to set uvb_params, overriding settings from passed-in UVBeam objects.
+
+    Raises
+    ------
+    ValueError :
+        For invalid beam_list, or if UVBeams in the input beam_list have
+        conflicting parameters (e.g., one has freq_interp_kind == 'cubic' and
+        another has freq_interp_kind == 'linear').
+
+    Notes
+    -----
+    When a UVBeam is appended to the beam list, its parameters override any in
+    uvb_params.
```

Comment:
Does this mean that my uvb_params on all my UVBeam objects are set by the last one I added? That seems a little perverse.

Reply:
Agreed. I've changed the behavior -- In object mode, conflicts in UVBeam parameters are allowed, but it will warn you if you try to add a UVBeam that conflicts. If you try to convert to string mode, it will error if there are conflicts.

Essentially, a BeamList in object mode behaves exactly like a list of beam objects -- it doesn't care what's on them -- but since differences in, e.g., `interpolation_function` will be lost by converting to strings, it won't let you if such differences exist.